schemas/v3.0/schema.yaml

```diff
@@ -0,0 +1,1400 @@
+         - string
+      not:
+        $ref: '#/definitions/Schema'
+      properties:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/Schema'
+      additionalProperties:
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - type: boolean
+        default: true
+      description:
+        type: string
+      format:
+        type: string
+      default: {}
+      $ref:
+        type: string
+        format: uri-ref
+      nullable:
+        type: boolean
+        default: false
+      discriminator:
+        $ref: '#/definitions/Discriminator'
+      readOnly:
+        type: boolean
+        default: false
+      writeOnly:
+        type: boolean
+        default: false
+      example: {}
+      externalDocs:
+        $ref: '#/definitions/ExternalDocumentation'
+      deprecated:
+        type: boolean
+        default: false
+      xml:
+        $ref: '#/definitions/XML'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false        
+  
+  Discriminator:
+    type: object
+    required:
+      - propertyName
+    properties:
+      propertyName:
+        type: string
+      mapping:
+        type: object
+        additionalProperties:
+          type: string
+
+  XML:
+    type: object
+    properties:
+      name:
+        type: string
+      namespace:
+        type: string
+        format: url
+      prefix:
+        type: string
+      attribute:
+        type: boolean
+        default: false
+      wrapperd:
+        type: boolean
+        default: false
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false
+
+  Response:
+    type: object
+    required:
+      - description
+    properties:
+      description:
+        type: string
+      headers:
+        additionalProperties:
+          oneOf:
+            - $ref: '#/definitions/Header'
+            - $ref: '#/definitions/Reference'
+      content:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/MediaType'
+      links:
+        type: object
+        additionalProperties:
+          oneOf:
+            - $ref: '#/definitions/Link'
+            - $ref: '#/definitions/Reference'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false
+
+  MediaType:
+    oneOf:
+      - $ref: '#/definitions/MediaTypeWithExample'
+      - $ref: '#/definitions/MediaTypeWithExamples'
+
+  MediaTypeWithExample:
+    type: object
+    properties:
+      schema:
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - $ref: '#/definitions/Reference'
+      example: {}
+      encoding:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/Encoding'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false 
+
+  MediaTypeWithExamples:
+    type: object
+    properties:
+      schema:
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - $ref: '#/definitions/Reference'
+      examples:
```

Comment:
Similarly, I had to make `examples` a required property to make the `oneOf` detection work correctly.

Reply:
@MikeRalphson 

You can also make things mutually exclusive by forbidding the other property rather than requiring the property that you have:

```YAML
oneOf:
  - properties:
        example:
            properties:
                ...
        examples:
            not: {}
  - properties:
        example:
            not: {}
        examples:
            patternProperties:
                ...
```